### DISCUSSION TITLE
Clarification on AqualinkD functionality

### DISCUSSION THREAD

**Discussion by Tim-K5AC** [aqualinkd/AqualinkD#335] [General]
Greetings!  I am a newbie pool owner and had a few questions about the AqualinkD project.   I just upgraded my Aqualink RS system from an old RS4 (Rev I) to an RS8 (Rev T.2) purchased on Ebay.  I am still using the All Touch RS4 wired remote, but it seems to be lacking some programming features.  I don't think that I can program the VSP from the All Touch wired remote, it appears that it can only be programmed from the One Touch remote ($600+) or the iAqualink3.0 ($500)??

Question: Can the AqualinkD/Raspberry PI REPLACE the All Touch wired remote or does the All Touch STILL have to be connected to the Aqualink RS panel?  

Question: If so, does this mean I can only run one protocol (All Touch or One Touch) on the AqualinkD?

I would like to use the AqualinkD/Raspberry PI to perform the functionality of the All-Touch/ iQ 3.0 and their web portal (iaqualink.com), I assume this is the primary purpose of AqualinkD?  I would like to have AqualinkD running on a Raspberry Pi4 with a Touch Screen Display mounted were the All Touch wired remote is/was located.

Thanks for the clarifications!

My System:
Aqualink RS8 (Rev T.2) with 6612F Power Center
Pentair Intelliflo VS Pump
Jandy JXI-400 Heater
All Touch RS-4 Wired Remote (located inside house)

Tim P.

**Comment by sfeakes** [comment]
The allbutton keypad will not program VSP and a few other things. The AqualinkD simulators will handle the programming for you, so no need to buy a $$$ remote, just use the AqualinkD OneTouch simulator.  With AqualinkD their is no need to have any keypads connected if you don’t want them, but it’s good to have one connected incase of issues. 
If you run serial_logger against the new panel, it will give you the best ID’s to use.  For a T.2 you will get 3 ID’s.  One for AlButton (required), one for RS-Serial (strongly recommended), once for either OneTouch or iAQtouch (needed for VSP).

Since you are moving from RS4 to RS8 make sure to reconfigure the buttons in `aqualinkd.conf` correctly.

Also make sure to read the wiki on the VSP pump ID’s and reading VSP / JXI information directly from the RS bus.

**Reply by Tim-K5AC** [reply]
Thanks for the response!  I have a Raspberry Pi4 and Touchscreen and FTDI-485 adapter on order.

I have a working RS4 combo+case, if you need it for development?  I'm in DFW area, where are you located?

**Reply by sfeakes** [reply]
@Tim-K5AC    Thank you very much for the offer, do you happen to know the panel version?  Usually it's a sticker on the main chip on the board.  If it's a version I don't have, then I would like to use it for development.  (I'm in Houston).

**Reply by ccruise13** [reply]
@Tim-K5AC I too just recently upgraded to the R8 from an R4 board. I haven't taken the time to attempt to reconfigure my AqualinkD.conf (I'm using my existing USB FTDI-485) but the prior configuration seemed to work as is with the new board. I might set aside some time this weekend and give it a go and will share my experience if you are interested.

**Reply by sfeakes** [reply]
@ccruise13  an AqualinkD RS4 board config will absolutely work on a physical RS8 panel, you will just loose 4 buttons.  If you changed the board rev# between the RS4 and RS8, you should run serial_logger as there may be new / faster features AqualinkD can use on the new board.
